### ISSUE TITLE
Default params aren't the same as on the Worldpay merchant area.

### ISSUE THREAD

**Issue by NoelDavies** [thephpleague/omnipay-worldpay#3]

**Comment by NoelDavies** [comment]
I soon realised this package uses the old Worldpay API. I'm writing up a new version for the new API and I'll submit it as a pull request soon.

**Comment by maclof** [comment]
I think this will be dependant on the age of the WorldPay account as I have access to 3 WorldPay accounts, none of which have a "service key" or "client key".

**Comment by NoelDavies** [comment]
@maclof I'm using the new API they have, I've written gateway for it, but due to the multiple gateways they have going, not sure how I should proceed with it contributing it back to Omnipay?

**Comment by maclof** [comment]
@NoelDavies Maybe this package should expose multiple gateways, one of them being the current one and the other being the new API? Similarly to how the PayPal package exposes 3 gateways, PayPal_Express, PayPal_Pro & PayPal_Rest - https://github.com/thephpleague/omnipay-paypal

**Comment by NoelDavies** [comment]
Yeah I've seen, but they haven't got different names for them... We could just name it Worldpay_Rest similarly to the paypal one?

**Comment by harrywebdev** [comment]
@NoelDavies have you got round to writing the new gateway?

**Comment by NoelDavies** [comment]
I have! But it's not fully tested! I'll update it on git later today and then post a link to it. I wrote it for an old client and they wouldn't let me write tests.. Yes.. one of those companies..

**Comment by andy1547** [comment]
@NoelDavis Any updates on this?

**Comment by JasonMortonNZ** [comment]
@NoelDavies - any progress on writing up the new gateway?

**Comment by bweston92** [comment]
Updates?

**Comment by JoSquiders** [comment]
I'd be interested in an update on this too please.

**Comment by lukeholder** [comment]
@NoelDavies please commit your work to github so we can contribute, otherwise I am starting on the new API gateway from scratch this week.

**Comment by dpfaffenbauer** [comment]
Updates?

**Comment by lukeholder** [comment]
@dpfaffenbauer I have made a pull request for the new api.

**Comment by lukeholder** [comment]
@JoSquiders @bweston92 @JasonMortonNZ @andy1547 @NoelDavies 

Please test my pull request and leave a comment on it if it's working for you.

Thanks.

**Comment by lukeholder** [comment]
This can also be closed @delatbabel

**Comment by jasegw** [comment]
Hi All.  This been fixed ?
If so, how do I fix please ?

My WorldPay details are...
Service key
Client key
WebHooks

OmniPay wants...
Installation Id
Account Id
Secret Word
Callback Password
